astropy/cosmology/funcs/comparison.py

```diff
@@ -0,0 +1,421 @@
+            raise ValueError(f"for a Cosmology, 'format' must be None or True, not {format}")
+        return cosmo
+
+    if format != False_:  # catches False and False_
+        format = None if format == True_ else format  # str->str, None/True/True_->None
+        out = Cosmology.from_format(cosmo, format=format)  # this can error!
+    elif not isinstance(cosmo, Cosmology):
+        raise TypeError(f"if 'format' is False, arguments must be a Cosmology, not {cosmo}")
+    else:
+        out = cosmo
+
+    return out
+
+
+def _parse_formats(*cosmos: object, format: _FormatsT) -> ndarray:
+    """Parse Cosmology-like to |Cosmology|, using provided formats.
+
+    `format` is broadcast to match the shape of the cosmology arguments.
+    Note that the cosmology arguments are not broadcast against ``format``,
+    so it cannot determine the output shape.
+
+    Parameters
+    ----------
+    *cosmos : |Cosmology|-like
+        The objects to compare. Must be convertible to |Cosmology|, as
+        specified by the corresponding `format`.
+
+    Raises
+    ------
+    TypeError
+        If any in 'cosmos' is not a |Cosmology| and the corresponding 'format'
+        equals `numpy.False_`.
+    """
+    formats = np.broadcast_to(np.array(format, dtype=object), len(cosmos))
+    # parse each cosmo & format
+
+    # Have to deal with things that do not broadcast well.
+    # astropy.row cannot be used in an array, even if dtype=object
+    # and will raise a segfault when used in a ufunc.
+    towrap = (isinstance(cosmo, _CosmologyWrapper._cantbroadcast) for cosmo in cosmos)
+    wcosmos = [(c if not wrap else _CosmologyWrapper(c)) for c, wrap in zip(cosmos, towrap)]
+
+    return _parse_format(wcosmos, formats)
+
+
+def _comparison_decorator(pyfunc: Callable[..., Any]) -> Callable[..., Any]:
+    """
+    Decorator to make wrapper function that parses cosmology-like inputs.
+
+    Parameters
+    ----------
+    pyfunc : Python function object
+        An arbitrary Python function.
+
+    Returns
+    -------
+    callable[..., Any]
+        Wrapped `pyfunc`, as described above.
+
+    Notes
+    -----
+    All decorated functions should add the following to 'Parameters'.
+
+    format : bool or None or str or tuple thereof, optional keyword-only
+        Whether to allow the arguments to be converted to a |Cosmology|.
+        This allows, e.g. a |Table| to be given instead a Cosmology.
+        `False` (default) will not allow conversion. `True` or `None` will,
+        and will use the auto-identification to try to infer the correct
+        format. A `str` is assumed to be the correct format to use when
+        converting.
+        `format` is broadcast to match the shape of the cosmology arguments.
+        Note that the cosmology arguments are not broadcast against ``format``,
+        so it cannot determine the output shape.
+    """
+    sig = inspect.signature(pyfunc)
+    nin = len([p.kind == 0 for p in sig.parameters.values()])
```

Comment:
``p.kind == 0`` is a filter for positional-only keywords. Just ``inspect.signature(pyfunc).parameters`` includes kwargs as well.

Reply:
You are creating a `list` of Booleans, but the length of that `list` does not depend on how many elements are `True` or `False`. It looks like you should be using `sum()` here instead of `len()`

Reply:
Oops. Thanks for the catch!